pennylane/templates/utils.py

```diff
@@ -76,9 +77,26 @@ def _get_shape(inpt):
     Returns:
         tuple: shape of ``inpt``
     """
-    inpt = np.array(inpt)
 
-    return inpt.shape
+    inpt = deepcopy(inpt)
```

Comment:
How come the deep copy is required?

Reply:
You are right, I cannot think of a situation where the function actually changes an object in python...deleted it